crates/utils/src/config/metrics.rs

```diff
@@ -0,0 +1,153 @@
+use getset::Getters;
+use serde::Deserialize;
+
+/// Xline metrics configuration object
+#[allow(clippy::module_name_repetitions)]
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Getters)]
+#[non_exhaustive]
+pub struct MetricsConfig {
+    /// Enable or not
+    #[getset(get = "pub")]
+    #[serde(default = "default_metrics_enable")]
+    pub enable: bool,
+    /// The http port to expose
+    #[getset(get = "pub")]
+    #[serde(default = "default_metrics_port")]
+    pub port: u16,
+    /// The http path to expose
+    #[getset(get = "pub")]
+    #[serde(default = "default_metrics_path")]
+    pub path: String,
+    /// Enable push or not
+    #[getset(get = "pub")]
+    #[serde(default = "default_metrics_push")]
+    pub push: bool,
+    /// Push endpoint
+    #[getset(get = "pub")]
+    #[serde(default = "default_metrics_push_endpoint")]
+    pub push_endpoint: String,
+    /// Push protocol
+    #[getset(get = "pub")]
+    #[serde(with = "protocol_format", default = "default_metrics_push_protocol")]
+    pub push_protocol: MetricsPushProtocol,
```

Comment:
Actually, I prefer to change the visibility of fields rather than use a public getter in that both achieve the same result, but the getter method introduces additional dependencies.